scrunch/datasets.py

```diff
@@ -1207,11 +1207,84 @@ def rollup(self, variable_alias, name, alias, resolution, description='',
         # return the variable instance
         return self[new_var['body']['alias']]
 
+    def derive_multiple_response(self, categories, subvariables, name, alias,
+        description='', notes=''):
+        """
+        This is the generic approach to create_multiple_response but this
+        allows the definition of any set of categories and rules (expressions)
+        for it's subvariables to fit these defined categories.
+        :param categories: a list of category dictionary:
+            categories=[{id: 1, name: 'Yes', 'selected': True} ...]
+        :param subvariables: a list of dictionary definind rules and attributes for each subvariable:
+            subvariables=[
+                {
+                    'id': 1,
+                    'name': 'Subvar 1',
+                    'cases': {
+                        1: 'var_1 < 20', 2: 'var_1 == 20', 3: 'var_1 == 30', 4: 'var_1 > 30'
+                    }
+                }
+            ]
+        :param name: Name of the variable to create
+        :param alias: Alias of the variable to create
+        :param description: Description of the variable to create
+        :param notes: Notes of the variable to create
+
+        Note: It's important that each subvariable defines the same numbes of cases for every category.
+        Suvariable alias will be derived from the alias to the variable + subvariable id to keep
+        compliance with other methods in Scrunch
+        """
+        # validate categories and fill defaults
+        categories = validate_categories(categories)
+        # validate that every subvar defines rules for all categories
+        for subvar in subvariables:
+            _validate_category_rules(categories, subvar['cases'])
+
+        responses_map = collections.OrderedDict()
+        responses_map_ids = []
```

Comment:
Why do you need this separate list? Could you just use `responses_map.keys()`, since it's an ordered dict, its keys will also be ordered.